pkg/config/validation/validation.go

```diff
@@ -2777,6 +2777,15 @@ var ValidateServiceEntry = RegisterValidateFunc("ValidateServiceEntry",
 			}
 		}
 
+		// check for v2 auto IP allocation or opt-out by user
```

Comment:
I'm assuming we have identical logic elsewhere. Can we rely on a util function?

Reply:
ooof, that produces a cyclic import so I'll need to circle back on that

Reply:
I moved the location of the helper functions to a new location in order to resolve the cyclic imports. @stevenctl, Can you take a look at if where I landed it is appropriate?